packages/gatsby-plugin-clerk/src/ssr/withServerAuth.ts

```diff
@@ -16,23 +16,28 @@ interface WithServerAuth {
 
 export const withServerAuth: WithServerAuth = (cbOrOptions: any, options?: any): any => {
   const callback = typeof cbOrOptions === 'function' ? cbOrOptions : undefined;
-  const opts = (options ? options : typeof cbOrOptions !== 'function' ? cbOrOptions : {}) || {};
+  const opts: WithServerAuthOptions = (options ? options : typeof cbOrOptions !== 'function' ? cbOrOptions : {}) || {};
 
   return async (props: GetServerDataProps) => {
     const req = gatsbyPropsToRequest(props);
     const requestState = await clerkClient.authenticateRequest(req, {
       ...opts,
       secretKey: SECRET_KEY,
       publishableKey: PUBLISHABLE_KEY,
-      request: req,
     });
 
     if (requestState.status === AuthStatus.Handshake) {
       // @TODO handle handshake
       return;
     }
 
-    const contextWithAuth = injectAuthIntoContext(props, requestState.toAuth());
+    const authObject = requestState.toAuth();
+    const contextWithAuth = await decorateObjectWithResources(
+      Object.assign(props, { auth: authObject }),
```

Comment:
Why are we adding the authObject twice?

Reply:
We decorate `props` with `auth` and then we use `auth` to load the resources and decorate the `props` obj with them, resulting in the following API:

```
props.auth
prop.user
props.session
props.organization
```

Let me rewrite this part for clarity

Reply:
Done!